### ISSUE TITLE
Allow user to disable SSL for Electrum server

### ISSUE THREAD

**Issue by siraben** [simpleledger/Electron-Cash-SLP#131]
In the network settings, there's a toggleable option for whether to use SSL or not, but the helpbox says that we can only disable SSL when using a Tor proxy. Instead, the option to disable SSL should be given no matter what for testing purposes.

**Comment by cculianu** [comment]
TBH I agree with you here.  I hate that it's locked.  @imaginaryusername and others wanted me to keep it locked.  I see Electron Cash as a power wallet for power users -- if you want an Apple-style wallet use something else.

So in that philosophy I 100% agree with you.  Sooo.. @imaginaryusername and @jcramer  @blockparty-sh .. what do you guys think about this?  I think we should let the user do what he/she wants.

Or is non-SSL that dangerous?

**Comment by imaginaryusername** [comment]
Since there's a "backdoor" where you can enable Tor, disable Tor, then viola you can use non-SSL as you wish... yeah, might as well let people do it all the time.

**Comment by cculianu** [comment]
We could just make them agree to an agreement clicking "I agree" like Apple does and then feel content that we did our due diligence.

**Comment by blockparty-sh** [comment]
Agree. Instead of some popup isn't it sort of obvious what disabling encryption means?

**Comment by cculianu** [comment]
IDK .. to me it is.  *shrug* Whatever people want. Popup, no popup. I'm ok with whatever.  I'm deep in the rabbit hole so to me it's very clear but -- maybe @imaginaryusername is more "in tune" with the common everyman.

**Comment by siraben** [comment]
What sort of changes would it require? In a local copy I was able to disable the SSL checkbox but it didn't seem to have any effect.

**Comment by imaginaryusername** [comment]
I'd say as long as the default is "Use SSL", we don't need the popup or disclaimer - option is sufficiently buried and risk not immediate enough, people can be responsible for their own decisions.

**Comment by cculianu** [comment]
@siraben How did you disable the checkbox exactly? It's not enough to kill the GUI element -- one must also walk into mordor and change the config key...

**Comment by siraben** [comment]
@cculianu I set `ssl_disable` at https://github.com/simpleledger/Electron-Cash-SLP/blob/e8a928243cc33a7672c1c37b217762a6cbb2613a/gui/qt/network_dialog.py#L735 to False, obviously it didn't work.

We're going to go with using SSL for now, even though that complicates our setup more. In the future it would be nice to not use SSL.
